internal/generator/vector/sources_to_inputs.go

```diff
@@ -74,7 +81,31 @@ func SourcesToInputs(spec *logging.ClusterLogForwarderSpec, o generator.Options)
 		}
 		el = append(el, r)
 	}
-	//TODO add user defined routing
+	//TODO add labels based routing
+	userDefined := spec.InputMap()
+	for _, pipeline := range spec.Pipelines {
+		for _, inRef := range pipeline.InputRefs {
+			if input, ok := userDefined[inRef]; ok {
+				// user defined input
+				if input.Application != nil {
+					app := input.Application
+					if len(app.Namespaces) != 0 {
+						matchNS := []string{}
+						for _, ns := range app.Namespaces {
+							matchNS = append(matchNS, fmt.Sprintf(IsNamespaceLog, ns))
+						}
+						el = append(el, Route{
+							ComponentID: RouteApplicationLogs,
+							Inputs:      helpers.MakeInputs("application"),
```

Comment:
nit.  dont we have a logtype constant for "application"

Reply:
actually, this is not a log type, this is name of the config block which generates application logs.
May be i can add another constant

Reply:
Seems like you could reasonably use the same constant, since the config block is related to the log type.